### PR TITLE
fix: implemented update to TXN_DATA

### LINKED ISSUE #1317: Update TXN_DATA
spec update : https://github.com/Consensys/linea-specification-internal/issues/614

### PR DESCRIPTION
**Pull request by OlivierBBB**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java [hunk 1]
```diff
@@ -75,47 +76,61 @@ public TxndataOperation(Wcp wcp, Euc euc, TransactionProcessingMetadata tx) {
...
+    callsToEucAndWcp.add(
+        TxnDataComparisonRecord.callToLt(accruedRefunds, refundLimit, getFullRefund));
+
+    // row 5: detecting empty payload
+    final Bytes payloadSize = Bytes.minimalBytes(tx.getBesuTransaction().getPayload().size());
+    final boolean payloadIsEmpty = wcp.callISZERO(payloadSize);
+    callsToEucAndWcp.add(TxnDataComparisonRecord.callToIszero(payloadSize, payloadIsEmpty));
+
+    // row 6: comparing the maximal gas price against the base fee
+    TransactionType type = tx.getBesuTransaction().getType();
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: could be final

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java [hunk 2]
```diff
@@ -75,47 +76,61 @@ public TxndataOperation(Wcp wcp, Euc euc, TransactionProcessingMetadata tx) {
...
+    final Bytes gasConsumedByTransactionExecution =
         Bytes.minimalBytes(tx.getBesuTransaction().getGasLimit() - tx.getLeftoverGas());
-    final Bytes row2arg2 = Bytes.of(MAX_REFUND_QUOTIENT);
-    final Bytes refundLimit = euc.callEUC(row2arg1, row2arg2).quotient();
-    callsToEucAndWcp.add(TxnDataComparisonRecord.callToEuc(row2arg1, row2arg2, refundLimit));
-
-    // i + effective_refund_row_offset
-    final Bytes refundCounterMax = Bytes.minimalBytes(tx.getRefundCounterMax());
-    final boolean getFullRefund = wcp.callLT(refundCounterMax, refundLimit);
+    final Bytes maxRefundQuotient = Bytes.of(MAX_REFUND_QUOTIENT);
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: could make a static BYTES_MAX_REFUND_QUOTIENT

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java [hunk 3]
```diff
@@ -75,47 +76,61 @@ public TxndataOperation(Wcp wcp, Euc euc, TransactionProcessingMetadata tx) {
...
+    // row 5: detecting empty payload
+    final Bytes payloadSize = Bytes.minimalBytes(tx.getBesuTransaction().getPayload().size());
+    final boolean payloadIsEmpty = wcp.callISZERO(payloadSize);
+    callsToEucAndWcp.add(TxnDataComparisonRecord.callToIszero(payloadSize, payloadIsEmpty));
+
+    // row 6: comparing the maximal gas price against the base fee
+    TransactionType type = tx.getBesuTransaction().getType();
+    final Bytes baseFee = Bytes.minimalBytes(tx.getBaseFee());
+    if (type == TransactionType.FRONTIER || type == TransactionType.ACCESS_LIST) {
+      Bytes gasPriceBytes =
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: final

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java [hunk 4]
```diff
@@ -75,47 +76,61 @@ public TxndataOperation(Wcp wcp, Euc euc, TransactionProcessingMetadata tx) {
...
+
+    // row 5: detecting empty payload
+    final Bytes payloadSize = Bytes.minimalBytes(tx.getBesuTransaction().getPayload().size());
+    final boolean payloadIsEmpty = wcp.callISZERO(payloadSize);
+    callsToEucAndWcp.add(TxnDataComparisonRecord.callToIszero(payloadSize, payloadIsEmpty));
+
+    // row 6: comparing the maximal gas price against the base fee
+    TransactionType type = tx.getBesuTransaction().getType();
+    final Bytes baseFee = Bytes.minimalBytes(tx.getBaseFee());
+    if (type == TransactionType.FRONTIER || type == TransactionType.ACCESS_LIST) {
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: could delete prefix `TransactionType.`

File: linea-constraints
Comment (letypequividelespoubelles) [inline linea-constraints:1]: it seems it doesn't contain https://github.com/Consensys/linea-constraints/pull/406 as it's not merge yet in master ? I'm surprised it passes the constraints then ...

**Review by letypequividelespoubelles — APPROVED** [review]
Only cosmetic comment

File: linea-constraints
Comment (OlivierBBB) [reply linea-constraints:1]: It does, see change to arithmetization/src/main/java/net/consensys/linea/zktracer/types/TransactionProcessingMetadata.java

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java (hunk 1 above)
Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: Done

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java (hunk 2 above)
Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: Done

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java (hunk 3 above)
Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: Done

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java (hunk 4 above)
Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/txndata/TxndataOperation.java]: Done

**Pull request merged**